enclave/src/nonce_handler.rs

```diff
@@ -0,0 +1,56 @@
+use log::*;
+use polkadex_sgx_primitives::{AccountId, AssetId};
+use sgx_tstd::collections::hash_map::HashMap;
+use sgx_tstd::collections::vec_deque::VecDeque;
+use sgx_tstd::vec::Vec;
+use sgx_types::{sgx_status_t, SgxResult};
+use std::sync::{
+    atomic::{AtomicPtr, Ordering},
+    Arc, SgxMutex, SgxMutexGuard,
+};
+
+static GLOBAL_POLKADEX_NONCE_STORAGE: AtomicPtr<()> = AtomicPtr::new(0 as *mut ());
+
+pub struct NonceHandler {
+    pub nonce: u32,
+    pub is_initialized: bool,
+}
+
+impl NonceHandler {
+    pub fn create() -> Self {
+        Self {
+            nonce: 0u32, //We can also use option
+            is_initialized: false,
+        }
+    }
```

Comment:
Suggestion: Add an increment function.

Then you could use nonce_storage.increment(); instead of  nonce_storage.nonce += 1; 
and write an unit test for it.

Reply:
done